File: simple-stack-example-multistack/src/main/java/com/zhuinden/simplestackdemomultistack/presentation/paths/main/list/ListKey.java
package com.zhuinden.simplestackdemomultistack.presentation.paths.main.list;

import android.os.Parcelable;

import com.google.auto.value.AutoValue;
import com.zhuinden.simplestackdemomultistack.R;
import com.zhuinden.simplestackdemomultistack.application.Key;
import com.zhuinden.simplestackdemomultistack.application.MainActivity;

/**
 * Created by Zhuinden on 2017.02.19..
 */

@AutoValue
public abstract class ListKey
        extends Key {
    @Override
    public int layout() {
        return R.layout.path_list;
    }

    public static Parcelable create() {
        return new AutoValue_ListKey();
    }

    @Override
    public String stackIdentifier() {
        return MainActivity.StackType.LIST.name();
    }
}

File: samples/legacy-samples/simple-stack-example-multistack-view/src/main/java/com/zhuinden/simplestackdemomultistack/features/main/list/ListKey.kt
package com.zhuinden.simplestackdemomultistack.features.main.list

import com.zhuinden.simplestackdemomultistack.R
import com.zhuinden.simplestackdemomultistack.application.MainActivity
import com.zhuinden.simplestackdemomultistack.core.navigation.MultistackViewKey
import kotlinx.parcelize.Parcelize

@Parcelize
data object ListKey : MultistackViewKey() {
    override fun layout(): Int = R.layout.list_view
    override fun stackIdentifier(): String = MainActivity.StackType.LIST.name
}

